audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/stream/SerDesConfig.java

```diff
@@ -132,15 +148,34 @@ private SerDesConfig() {
     /**
      * Kafka value deserialiser for upstream messages coming in as input
      *
+     * @param configProperties contains the directory for error files
      * @return an appropriate value deserialiser for the topic's message content (AuditSuccessMessage)
      */
-    public static Deserializer<AuditSuccessMessage> successValueDeserializer() {
+    public static Deserializer<AuditSuccessMessage> successValueDeserializer(final AuditServiceConfigProperties configProperties) {
         return (String ignored, byte[] auditRequest) -> {
             try {
                 return MAPPER.readValue(auditRequest, AuditSuccessMessage.class);
             } catch (IOException e) {
-                throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + new String(auditRequest, Charset.defaultCharset()), e);
+                String failedAuditString = new String(auditRequest, Charset.defaultCharset());
+                createFile("Success-", failedAuditString, configProperties);
+                SerDesHealthIndicator.addSerDesExceptions(e);
+                throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + failedAuditString, e);
             }
         };
     }
+
+    private static void createFile(final String prefix, final String failedAuditString, final AuditServiceConfigProperties configProperties) {
+        String fileName = prefix + ZonedDateTime.now(ZoneOffset.UTC)
+                .format(DateTimeFormatter.ISO_INSTANT).replace(":", "-");
+        File directory = new File(configProperties.getErrorDirectory());
```

Comment:
If the user doesnt have write permissions over the errorDirectory, they'll get a `java.io.IOException: The filename, directory name, or volume label syntax is incorrect` sort of error, again, might be worth in the readme, including common faults and solutions (permissions, environment specific names)

Reply:
Not sure this is relevant as the value should be configured in the `application.yaml`. It might be something to speak about later.